realm/realm-library/src/objectServer/java/io/realm/PermissionManager.java

```diff
@@ -1095,11 +1095,8 @@ public void onChange(RealmResults<PermissionOffer> loadedOffers) {
         }
 
         void notifyCallbackWithSuccess(RealmResults<PermissionOffer> permissions) {
-            try {
-                callback.onSuccess(permissions);
-            } finally {
-                activeTasks.remove(this);
-            }
+            activeTasks.remove(this);
```

Comment:
maybe just using try-finally in the other cases would have also fixed the problem if the order matters for some reason